src/main.cpp

```diff
@@ -19,6 +19,8 @@ int main(int argc, char** argv) {
     for (auto i = 0; i < argc; ++i)
         cmd += std::string(i == 0 ? "" : " ") + argv[i];
     setenv("HYPRLAND_CMD", cmd.c_str(), 1);
+    setenv("XDG_BACKEND", "wayland", 1);
+    setenv("_JAVA_AWT_WM_NONREPARENTING", "1", 1);
```

Comment:
should we overwrite them if the user specified something else?

also wtf is xdg backend

Reply:
for the XDG one, probably since it should always be wayland for wayland compositors, the java one I don't know though, it's probably ok to not override that, it's not set by i3 or xfce, don't know about sway, but that seems pretty particular.
Do you know anything about this var @fufexan?

Reply:
It's apparently important for java swing apps: https://github.com/xmonad/xmonad/issues/126
it's probably ok to overwrite this as well, it needs to be anything else that not defined: https://bugs.openjdk.org/browse/JDK-8058197
so if a user defined it and we overwrite it with a 1, it shouldn't be a big deal I guess, as you want tbh

Reply:
I've changed it to not overwrite to be safe